### DISCUSSION TITLE
Constructing a covariate model

### DISCUSSION THREAD

**Discussion by Mannny-PK** [LAPKB/Pmetrics#136] [Q&A]
Dear Professor, 
Hello! I would like to ask a question: which indicator should I choose for model selection when constructing a covariate model that I want to use the stepwise method i.e. forward inclusion and backward exclusion: -2LL, AIC or BIC?

**Comment by mhovd** [comment]
The AIC and BIC are just different methods to incorporate the log likelihood with regards to the number of parameters or the number of subjects. 

$\text{AIC} = -2\ln(L) + 2k$
$\text{BIC} = -2\ln(L) + k\ln(n)$

Where $ln(L)$ is the `-2LL`, `k` the number of parameters in the model and $n$ the number of subjects.
If the number of parameters and the number of subjects is the same, it does not matter which metric you use, as they only rely on `-2LL`.

**Comment by ModelingDegu** [comment]
Hi Manny
If you have a stable minimization routine: Full Model forward, i.e., all the covariates you can think of as relevant, then after convergence, only backward elimination using the criteria above. Full fixed effects model is the name of the approach. Consider to build a base model, estimate the dose for every single individual for target attainment and do your covariate search on the dose. That is what you see in the label, the covariate-"relevant dose" relationship. Takes away a lot of the dimensionality of the search space and gives you the answer you really need.

**Comment by mnneely** [comment]
I think an interesting possible workflow is to incorporate machine learning into the process of covariate selection. I’m just learning about this myself, but here’s a [PubMed search](https://pubmed.ncbi.nlm.nih.gov/?term=Machine+learning+covariate+Pharmacometrics) with several relevant articles. We haven’t incorporated this into Pmetrics yet, but I imagine that we will include something eventually!
